### PR TITLE
feat: migrate to xterm.js 5.3.0

### LINKED ISSUE #7: Upgrade to xterm.js 4.14 when it ships
Just came across this, very cool 🙂 

FYI we recently implemented custom glyphs for box drawing and block elements which will make your readme images much more fancy.

image
image

I'm going to try get the release done this week sometime.

### PR DESCRIPTION
**Pull request by edouard-lopez**
- refactor: split `lint` script into `lint`, `compile` and `test
- chore: update packages setup eslint and simplify prettier config as we use default
- feat(xterm): migrate to 5.3.0

### CODE REVIEW

**Review by OmarTawfik — CHANGES_REQUESTED** [review]
@edouard-lopez thank you so much for looking into the xterm migration!
However, I suggest not mixing this upgrade with other drive-by changes, that are not required/related.

Thoughts about keeping this PR minimal to just the upgrade? we can discuss any infra/tooling related changes separately. Afterwards, I would be happy to test it and release a new version.

Thanks!

**Comment by edouard-lopez** [comment]
I restrict the MR to a commit bumping `xterm` and the related update. I will push another MR for the tooling

Do you have constraints on the tooling versions? Do you have external tests?

**Review by OmarTawfik — CHANGES_REQUESTED** [review]
Thank you for looking into this!

I pulled it and ran the tests locally. Existing tests (comparing against static images) are passing except one:

```log

  ● can render error-message

    Expected image to match or be a close match to snapshot but was 11.925925925925926% different from snapshot (644 differing pixels).
    See diff for details: /var/folders/wf/m183hpcj025f1psz6nlsl83r0000gn/T/error-message-diff.png
```

And here is the diff:

error-message-diff

I wonder if it is passing for you?
Maybe we just need to update the static images, as there might be glyph/character differences because of the upgrade.

**Comment by edouard-lopez** [comment]
I checked again by rebuild the one from `main` from a clean slate:

```
❯ git checkout main
❯ rm yarn.lock node_modules/ -rf
❯ rm tests/screenshots/* 
❯ yarn install --frozen-lockfile
❯ jest tests/screenshots.test.ts
```

Then back on the feature branch
```
❯ git checkout feat/migrate-to-xterm.js-5.3.0
❯ rm yarn.lock node_modules/ -rf
❯ yarn install --frozen-lockfile
❯ jest tests/screenshots.test.ts
```

All are failing
white-background-diff
emojis-diff
error-message-diff
custom-background-diff
custom-margin-diff

Looking closer, the `!` (exclamation point) and `.` final dot are missing

**Comment by edouard-lopez** [comment]
## Feature Branch

The template I got for the `error-message` test:

```html
<div class="xterm-rows" style="line-height: normal; letter-spacing: 0px;" aria-hidden="true">
    <div style="width: 270px; height: 17px; line-height: 17px; overflow: hidden;">
        <span class="xterm-fg-1">ERROR:</span>
        <span class="xterm-fg-8"> something went wrong!!</span>
        <span class="xterm-cursor xterm-cursor-outline xterm-fg-8">!</span>
    </div>
</div>
```

There is something weird happening with the cursor it seems
image

##  Main branch

Template render a `<canvas>` element. The [5.0.0](https://github.com/xtermjs/xterm.js/releases/tag/5.0.0) switch away from canvas for rendering (look for _Canvas renderer addon_ section)
image

**Comment by edouard-lopez** [comment]
@OmarTawfik Could you check again, I believe it was due to the way you hide the cursor that was not behaving in the same way with `5.3.0`. 

I set [`cursorInactiveStyle: none`](https://github.com/xtermjs/xterm.js/releases/tag/5.3.0)  introduced in `5.3.0` instead

**Comment by OmarTawfik** [comment]
@edouard-lopez  Thank you for the investigation. I did some digging and discovered a couple of issues (fixed in #16). Please feel free to rebase, and it should be able to run/test successfully now.

**Comment by edouard-lopez** [comment]
Rebased and fixed a bug in CI, due to @ts-node

File: yarn.lock [hunk 1]
```diff
@@ -2,302 +2,303 @@
...
-    "@babel/generator" "^7.14.0"
-    "@babel/helper-compilation-targets" "^7.13.16"
-    "@babel/helper-module-transforms" "^7.14.0"
-    "@babel/helpers" "^7.14.0"
-    "@babel/parser" "^7.14.0"
-    "@babel/template" "^7.12.13"
-    "@babel/traverse" "^7.14.0"
-    "@babel/types" "^7.14.0"
-    convert-source-map "^1.7.0"
+"@ampproject/remapping@^2.2.0":
```
Comment (OmarTawfik) [inline yarn.lock]: This PR changes the entire `yarn.lock` from `yarnpkg` registry to `npmjs` .. I wonder if you have custom local overrides (e.g. `yarnrc`). If not, I suggest reverting these changes, and undoing changes to this file and rerunning `yarn` with vanilla settings.

**Comment by edouard-lopez** [comment]
I rebased following #14 merge

**Review by OmarTawfik — APPROVED** [review]
Looks great!
Thank you.

**Pull request merged**